zebedee-cms/src/test/java/com/github/onsdigital/zebedee/util/slack/SlackNotifierTest.java

```diff
@@ -0,0 +1,576 @@
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class SlackNotifierTest {
+    @Mock
+    private SlackClient slackClient;
+
+    @Mock
+    private PostMessageResponse messageResponse;
+
+    @Mock
+    private PostMessage pm;
+
+    @Mock
+    private Collection collection;
+
+    @Mock
+    private CollectionDescription description;
+
+    @Mock
+    private Profile profile;
+
+    @Mock
+    private AttachmentField field;
+
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void sendSlackMessageNil() throws Exception {
+        when(slackClient.sendMessage(pm))
+                .thenReturn(messageResponse);
+
+        SlackNotifier sn = new SlackNotifier(slackClient);
+        sn.sendSlackMessage(pm);
+
+        verify(slackClient, times(1)).sendMessage(any());
+    }
+
+    @Test
+    public void sendSlackMessageException() {
+        when(slackClient.getProfile())
+                .thenReturn(profile);
+        when(slackClient.sendMessage(pm))
+                .thenThrow(RuntimeException.class);
+
+        SlackNotifier sn = new SlackNotifier(slackClient);
+        Exception actual = assertThrows(Exception.class, () -> sn.sendSlackMessage(pm));
+        assertThat(actual.getMessage(), equalTo("unexpected error sending slack message"));
+        verify(slackClient, times(1)).sendMessage(pm);
+    }
+
+    @Test
+    public void sendCollectionAlarmException() {
+        PostMessage pm = new PostMessage("username", "channel", ":grinning:", "yes");
+
+        when(collection.getDescription())
+                .thenReturn(description);
+        when(description.getType())
+                .thenReturn(manual);
+        when(description.getName())
+                .thenReturn("Test");
+        when(collection.getId())
+                .thenReturn("101");
+        when(slackClient.getProfile())
+                .thenReturn(profile);
+        when(profile.newPostMessage(anyString(), anyString()))
+                .thenReturn(pm);
+        when(slackClient.sendMessage(pm))
+                .thenThrow(RuntimeException.class);
+
+        SlackNotifier sn = new SlackNotifier(slackClient);
+        boolean returnValue = sn.sendCollectionAlarm(collection, "Any", "custom", new Exception("Hello test"));
+
+        verify(slackClient, times(1)).sendMessage(pm);
+        assertFalse(returnValue);
+    }
+
+
+    @Test
+    public void sendCollectionAlarmChannelEmpty() {
+        SlackNotifier sn = new SlackNotifier(slackClient);
+        boolean returnValue = sn.sendCollectionAlarm(collection, "", "message", new Exception("Hello test"));
+
+        verify(slackClient, times(0)).sendMessage(pm);
+        assertThat(pm.getAttachments().size(), equalTo(0));
+        assertFalse(returnValue);
+    }
```

Comment:
`channel` can also be `null`. Duplicate this test but pass in `null` rather than `""`.

I know this feels possibly over the top _but_ it is possible that `null` could be provided and that will affect how this code behaves therefore we should confirm the behaviour with a test to prove it works as we expect.

Same for `message` and `exception` - add test cases to prove it behaves as expected if they are null.

Reply:
I have added the test check ex.getMessage() to be null or empty. Not sure, if I should be checking the exception itself rather than ex.getMessage()